microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/keyselection/MultipleConfiguredPublicKeysSelectionTest.java

```diff
@@ -0,0 +1,212 @@
+                .then()
+                .statusCode(401);
+    }
+
+    /**
+     * @tpTestDetails A JWT signed by a "orange" private key is send to the server which has configured public key to be
+     *                a JSON Web Key set Base64 encoded. There are multiple keys in this set and correct one must be chosen
+     *                based on {@code kid} value in JOSE header of JWT.
+     * @tpPassCrit Correct public key from the JWKS is chosen and client receives token raw value in response.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_WITH_BASE64_JWKS)
+    public void testJwtSignedByOrangeKeyBase64(@ArquillianResource URL url) {
+        JsonWebToken token = new JwtHelper(rsaKeyToolOrange, "issuer").generateProperSignedJwt();
+
+        given().header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "secured-endpoint")
+                .then()
+                .statusCode(200)
+                .body(equalTo(token.getRawValue()));
+    }
+
+    /**
+     * @tpTestDetails A JWT signed by a "blue" private key is send to the server which has configured public key to be
+     *                a JSON Web Key set Base64 encoded. There are multiple keys in this set and correct one must be
+     *                chosen based on {@code kid} value in JOSE header of JWT.
+     * @tpPassCrit Correct public key from the JWKS is chosen and client receives token raw value in response.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_WITH_BASE64_JWKS)
+    public void testJwtSignedByBlueKeyBase64(@ArquillianResource URL url) {
+        JsonWebToken token = new JwtHelper(rsaKeyToolBlue, "issuer").generateProperSignedJwt();
+
+        given().header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "secured-endpoint")
+                .then()
+                .statusCode(200)
+                .body(equalTo(token.getRawValue()));
+    }
+
+    /**
+     * @tpTestDetails A JWT signed by a "pink" private key is send to the server which has configured public key to be a
+     *                JSON Web Key set Base64 encoded. There are multiple keys in this set and there is no "pink" public
+     *                key among them.
+     * @tpPassCrit JWt is rejected and user receives 401/forbidden because there is no matching configured public key on
+     *             the server.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_WITH_BASE64_JWKS)
+    public void testJwtSignedByPinkKeyBase64Pk(@ArquillianResource URL url) {
+        JsonWebToken token = new JwtHelper(rsaKeyToolPink, "issuer").generateProperSignedJwt();
+
+        given().header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "secured-endpoint")
+                .then()
+                .statusCode(401);
```

Comment:
Looking at the tests there is checked either 200 or 401 is returned. It's repeated in the tests in test case. Do you think that you might hava just 2 methods which checks 200 or 400 do not duplicate the rest call?

Reply:
It is possible to reduce code redundancy. But is it really worth it in this case? Right now, it is clear what exactly happens in each test without having to read other methods. I can do it, but I don;t think it's necessary.

Reply:
Thinking of it, it might be tough use 2 assertions methods in the future. Let's keep as it is and have tests independent.